### PR TITLE
feat: add docker image for api

### PR DESCRIPTION
**Pull request by 0xkenj1**
# 🤖 Linear

Closes ZKS-209

## Description
- Add Dockerfile

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Ignore all .env files in any directory or subdirectory
+**/.env
+
+# Ignore node_modules and dist directories in any directory or subdirectory
+**/node_modules
+**/dist
+
+**/*.md
```
Comment (0xyaco) [inline .dockerignore:8]: Don't forget to include some key folders:
* `.git`
* `.turbo`

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+RUN corepack enable
+
+FROM base AS build
+COPY . /usr/src/app
+WORKDIR /usr/src/app
+RUN --mount=type=cache,id=pnpm,target=/pnpm/store pnpm install --frozen-lockfile
+RUN pnpm run -r build
+RUN pnpm deploy --filter=./apps/api --prod /prod/api
+
+# RUN pnpm deploy --filter=app2 --prod /prod/app2
```
Comment (0xyaco) [inline Dockerfile:14]: Should these comments stay?

**Review by 0xyaco — COMMENTED** [review]
:building_construction:

**Review by 0xnigir1 — DISMISSED** [review]
nothing to add to yaco comments, lets go ⏭️ 

just for my curiosity, is there any difference between `pnpm deploy` vs [`turbo deploy`](https://turbo.build/repo/docs/guides/tools/docker)?

File: Dockerfile (hunk 2 above)
Comment (0xkenj1) [reply Dockerfile:14]: mmmm, yes imo. It might be helpful for anyone that may want to build an image for a new app.

**Comment by 0xkenj1** [comment]
not sure if `turbo deploy` is a command 🤣

**Review by 0xnigir1 — APPROVED** [review]

**Pull request merged**